js/core/transform-json-response.js

```diff
@@ -1,99 +1,72 @@
+  const page = new schema.Entity("pages", {});
   const question = new schema.Entity("questions", {}, {
-    idAttribute: (q) => q.key,
+    idAttribute: value => value.key
   });
-  // Answers don't have unique ID so generate it. Embeddable key + student ID works.
-  const answer = new schema.Entity("answers", {}, {
-    idAttribute: (a) => `${a.embeddableKey}-${a.studentId}`,
-  });
-  const feedback = new schema.Entity("feedbacks", {}, {
-    idAttribute: (a) => a.answerKey,
-  });
-
-  const activityFeedback = new schema.Entity("activityFeedbacks", {}, {
-    idAttribute: (value, parent) => `${parent.id}-${value.studentId}`,
-  });
-
-  investigation.define({
+  sequence.define({
     children: [ activity ],
   });
   activity.define({
     children: [ section ],
-    activityFeedback: [ activityFeedback ],
   });
   section.define({
     children: [ page ],
   });
   page.define({
     children: [ question ],
   });
-  question.define({
-    answers: [ answer ],
-  });
-  answer.define({
-    feedbacks: [ feedback ],
-  });
+  return normalize(camelizedJson, sequence);
+}
 
-  const reportType = camelizedJson.report.type;
-  const response = normalize(migrate(camelizedJson), {
-    report: reportType === "Investigation" ? investigation : activity,
-    "class": {
-      students: [ student ],
-    },
-  });
-  // Post-process response:
-  // Provide fake investigation if it's not present to simplify app logic.
-  if (reportType === "Activity") {
-    response.entities.investigations = {
-      1: {
-        id: 1,
-        name: "",
-        type: "Investigation",
-        children: [response.result.report],
-      },
+export function preprocessResourceJson(resourceJson) {
+  // Provide fake sequence if it's not present to simplify app logic.
+  if (resourceJson.type === "Activity") {
+    resourceJson = {
+      id: 1,
+      name: "",
+      type: "Sequence",
+      children: [ resourceJson ],
     };
   }
-  response.type = camelizedJson.reportFor || DEFAULT_REPORT_FOR;
-  response.hideControls = camelizedJson.hideControls || false;
-  applyVisibilityFilter(response.entities.questions, response.result.visibilityFilter);
-  copyAnswerKeysToObjects(response.entities.answers || []);
-  copyAnswerKeysToObjects(response.entities.activityFeedbacks || []);
-  saveStudentsRealNames(response.entities.students);
-  urlParamOverrides(response);
-  return response;
+  // Add some question properties, e.g. question numbers, selection, visibility.
+  resourceJson.children.forEach(activity => {
+    activity.children.forEach(section => {
+      // Questions are numbered within the whole activity.
+      let currentQuestionNumber = 1;
+      section.children.forEach(page => {
+        page.children.forEach(question => {
+          question.questionNumber = currentQuestionNumber++;
+          // Id is not enough, as questions of different type can have the same id.
+          question.key = `${question.type}-${question.id}`;
```

Comment:
Do you think we should create this id on export from LARA?

Reply:
I guess it's fine that we expect authoring system / LARA to publish things that have unique `type + id` combination and deal with that here (as it could also affect other containers with clashing IDs and different types).

But I'm wondering how will we match answers and questions. Have you thought about that in your LARA run export code? 
I guess it should be also something like this `key` here. Then, it could make sense for LARA to generate these keys both in questions and answers, so we can be sure that they match and reporting system doesn't have to care about that.

Reply:
I   think you are right: Lara should create this key in questions and answers (for matching  purposes)